src/core/core.controller.js

```diff
@@ -933,7 +937,65 @@ module.exports = function(Chart) {
 			me.lastActive = me.active;
 
 			return changed;
-		}
+		},
+		/**
+		 * format the data list if the data is missing some points in the datasets.
+		 * @private
+		 * @param {array} dataArray array to format
+		 * @return {array} the formated array
+		 */
+		formatDataset: function(dataArray) {
+			var labels = this.chart.data.labels;
+			var tmp = dataArray.slice(0);
+			var labelLen = labels.length;
```

Comment:
there's no need to create a variable for this value since it's only used once

Reply:
I have removed the  `labelLen` and `dataLen` variables since they are used only once.